address_gateway.go

```diff
@@ -39,3 +39,21 @@ func (g *AddressGateway) Delete(ctx context.Context, customerId, addrId string)
 	}
 	return &invalidResponseError{resp}
 }
+
+// Updates an address for the specified id and customer id.
+func (g *AddressGateway) Update(ctx context.Context, customerId, addrId string, a *Address) (*Address, error) {
+	// Copy address so that field sanitation won't affect original
+	var cp Address = *a
+	cp.CustomerId = ""
```

Comment:
Instead of sanitizing the address fields, I think we need to employ the same pattern here that we've moved to in other gateways, where there is an `AddressRequest` for creates and updates, and an `Address` for finds. I  can see this follows the same pattern right now as the create though.